test/attr/attr_escaping.swift

```diff
@@ -72,15 +72,14 @@ struct GenericStruct<T> {}
 
 func misuseEscaping(_ a: @escaping Int) {} // expected-error{{@escaping attribute only applies to function types}} {{26-36=}}
 func misuseEscaping(_ a: (@escaping Int)?) {} // expected-error{{@escaping attribute only applies to function types}} {{27-36=}}
-func misuseEscaping(opt a: @escaping ((Int) -> Int)?) {} // expected-error{{@escaping attribute only applies to function types}} {{28-38=}}
-  // expected-note@-1{{closure is already escaping in optional type argument}}
+func misuseEscaping(opt a: @escaping ((Int) -> Int)?) {} // expected-error{{closure is already escaping in optional type argument}} {{28-38=}}
 
 func misuseEscaping(_ a: (@escaping (Int) -> Int)?) {} // expected-error{{@escaping attribute may only be used in function parameter position}} {{27-36=}}
```

Comment:
I believe in this call and all cases below it we wouldn't actually produce a `@escaping attribute may only be used in function parameter position` diagnostic anymore.

Reply:
Unfortunately, we would. They produced from a different part of code, to be precise from 2387-2394 lines in TypeCheckType.cpp
I'm not sure that we should fix these cases too because it's correct to produce `@escaping attribute may only be used in function parameter position` here cause escaping inside closure declaration and it's an error too
But I can fix this behavior 🧐
@xedin